src/app/common/filter-text/filter-text.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, EventEmitter, Output } from '@angular/core';
+
+@Component({
+  selector: 'ipaas-filter-text',
+  template: '<input type="text" id="filterText" [(ngModel)]="filter" (keyup)="filterChanged($event)" />',
+})
+export class FilterTextComponent {
+  @Output() changed: EventEmitter<string>;
+  filter: string;
+
+  constructor() {
+    this.changed = new EventEmitter<string>();
+  }
+
+  clear() {
+    this.filter = '';
+  }
+
+  filterChanged(event: any) {
```

Comment:
We should be debouncing these events. How about using rxjs something like http://plnkr.co/edit/98ijjY?p=preview (see script.ts):

```typescript
const input$ = Rx.Observable
  .fromEvent(document.getElementById('hello'), 'keyup')
  .map(x => x.currentTarget.value)
  .debounceTime(500);
  
const updated$ = document.getElementById('updated');
input$.subscribe(x => updated.innerHTML = x);
```